azurerm/resource_arm_proximity_placement_group.go

```diff
@@ -0,0 +1,126 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmProximityPlacementGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmProximityPlacementGroupCreateUpdate,
+		Read:   resourceArmProximityPlacementGroupRead,
+		Update: resourceArmProximityPlacementGroupCreateUpdate,
+		Delete: resourceArmProximityPlacementGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Validate how? I'm not going to pretend I'm not rather new to this. Much of this is taken directly from availability sets since the requirements were very similar.